### PR TITLE
Adding third party library for struct 

### PR DESCRIPTION
**Pull request by badarsh2**
The purpose of adding this library is to provide headers for unpacking binary-style trajectories, particularly of .trr and .dcd formats.
Original source of the library: https://github.com/svperbeast/struct

### CODE REVIEW

**Comment by badarsh2** [comment]
Since the library is not in active development, I hope it's alright to have it as self-contained third party code instead of as an external dependency.

**Comment by cryos** [comment]
I have fixed the issue with submodules, so it will get further. We cannot merge in struct if we can't get it working on Windows, macOS, and Linux. You have appveyor builds so that you can try changes without merging them, I will try and take a look too but nothing obvious jumped out. Hopefully we can get to the bottom of it, if the library is not cross-platform/maintained we may want to look for other alternatives.

File: avogadro/io/CMakeLists.txt [hunk 1]
```diff
@@ -51,7 +52,7 @@ endif()
 
 avogadro_add_library(AvogadroIO ${HEADERS} ${SOURCES})
 
-target_link_libraries(AvogadroIO LINK_PUBLIC AvogadroCore LINK_PRIVATE jsoncpp)
+target_link_libraries(AvogadroIO LINK_PUBLIC AvogadroCore LINK_PRIVATE jsoncpp LINK_PRIVATE struct)
```
Comment (cryos) [inline avogadro/io/CMakeLists.txt]: You don't need to repeat LINK_PRIVATE, everything after will use private linking.

**Comment by badarsh2** [comment]
I tried building the supermodule using visual studio under "Release" config, and I was getting some errors from mongochem. 
image
image

**Comment by badarsh2** [comment]
The appveyor build seems to have failed not because of struct, but because of an issue with mongochem.

**Comment by cryos** [comment]
Are you sure you have the latest version? I just finished removing all of MongoChem from it, so any fresh checkout would not even have MongoChem in the repository.

**Comment by cryos** [comment]
The error `fatal: No url found for submodule path 'mongochem' in .gitmodules` should be resolved now, it is because I had forgotten one step in removing a submodule. Pushing a new commit/restarting the appveyor build would move beyond that error - in Git, not in the build.

**Comment by cryos** [comment]
I would advise Visual Studio 2017 with Win64, I have been testing that and it works here. Maybe Visual Studio 2015 will work too, but they are both free for open source development, so use the latest one that I am testing with. I was able to compile and link the latest, make sure you do a
```
git pull
git submodule update --recursive
```
That will ensure that MoleQueue etc are all up to date.

**Comment by badarsh2** [comment]
The build goes smooth after changing to Win64 instead of the x86 compiler. Also, I disabled BUILD_MOLEQUEUE as it was not getting built with x64 compilers. Now on building, I have some errors saying 

> error C2065: 'M_PI': undeclared identifier

How do I fix these?

**Comment by badarsh2** [comment]
Changing the compiler to WIn64 helped: I got rid of the errors I waa getting earlier. Now though, I'm getting a slew of errors in several of the core classes of avogadrolibs, saying: `error C2065: 'M_PI': undeclared identifier`. I'm unable to figure out what's going wrong. I've attached the whole log of the superbuild compilation. I'm using Qt5.11's msvc2015_64 config files and Visual Studio 2017, 
[log.txt](https://github.com/OpenChemistry/avogadrolibs/files/2074345/log.txt)

**Comment by badarsh2** [comment]
It would be really nice if you can help me with compiling the struct files and checking compatibility with Windows. I've been stuck with the installation for the past three days and I am having trouble figuring out the build errors.

**Comment by badarsh2** [comment]
I also realized that the appveyor build log for [this PR](https://github.com/OpenChemistry/avogadrolibs/pull/272#partial-pull-merging) also shows similar errors (https://ci.appveyor.com/project/ghutchis/avogadrolibs/build/1.0.70)

**Comment by cryos** [comment]
I will take a look, but pushing the few minor changes I asked for would have triggered a fresh build for this relatively clean branch (and address issues I pointed out that need changing anyway). I had  noticed those, and will try a clean build locally to see what is happening.

**Comment by badarsh2** [comment]
The build seems to have run without errors.
I also pulled the configuration changes, compiled struct with MSVC and tested it with the GROMACS trr readers I had written, and it seems to be working.

**Comment by badarsh2** [comment]
A gentle reminder to kindly review this PR, so that I can proceed with working on the other PRs that are dependent on this library.

**Pull request merged**